pkg/api/annotations.go

```diff
@@ -212,6 +214,8 @@ func PatchAnnotation(c *models.ReqContext, cmd dtos.PatchAnnotationsCmd) Respons
 		existing.EpochEnd = cmd.TimeEnd
 	}
 
+	existing.Data = cmd.Data
```

Comment:
Need an if statement to check if `cmd.Data` is provided.

Reply:
I was confused about this initially. I thought that if the user wants to clear the data from the annotation, then they would pass null for `cmd.Data`. But yeah, it makes sense to me now. Fixed this in the next commit.